modules/core/src/mathfuncs_core.simd.hpp

```diff
@@ -185,13 +110,60 @@ static inline void atanImpl(const T *Y, const T *X, T *angle, int len, bool angl
 void fastAtan32f(const float *Y, const float *X, float *angle, int len, bool angleInDegrees )
 {
     CV_INSTRUMENT_REGION()
-    atanImpl<float>(Y, X, angle, len, angleInDegrees);
+    float scale = angleInDegrees ? 1.f : (float)(CV_PI/180);
+    int i = 0;
+#if CV_SIMD
+    const int VECSZ = v_float32::nlanes;
+    v_atan_f32 v(scale);
+
+    for( ; i < len; i += VECSZ*2 )
+    {
+        if( i + VECSZ*2 > len )
+        {
+            // if it's inplace operation, we cannot repeatedly process
+            // the tail for the second time, so we have to use the
+            // scalar code
+            if( i == 0 || angle == X || angle == Y )
+                break;
+            i = len - VECSZ*2;
```

Comment:
if it's inplace operation and something is wrong then X/Y are already corrupted.

IMHO, vectorized loops in this file are malformed.
Proper design pattern is:
```
for( ; i <= len - BLOCK_SIZE; i += BLOCK_SIZE)
{
    // no more 'if's
    ...
}
for( ; i < len; i++ )
{
    ...
}
```